src/cassiopeia/critique/compare.py

```diff
@@ -129,25 +198,34 @@ def robinson_foulds(tree1: CassiopeiaTree, tree2: CassiopeiaTree) -> tuple[float
         The Robinson-Foulds distance between the two trees and the maximum
             Robinson-Foulds distance for downstream normalization
     """
-    # create copies of the trees and collapse process
-    T1 = copy.deepcopy(tree1)
-    T2 = copy.deepcopy(tree2)
-
-    # convert to Ete3 trees and collapse unifurcations
-    T1.collapse_unifurcations()
-    T2.collapse_unifurcations()
-
-    T1 = ete3.Tree(T1.get_newick(), format=1)
-    T2 = ete3.Tree(T2.get_newick(), format=1)
-
-    (
-        rf,
-        rf_max,
-        names,
-        edges_t1,
-        edges_t2,
-        discarded_edges_t1,
-        discarded_edges_t2,
-    ) = T1.robinson_foulds(T2, unrooted_trees=True)
-
-    return rf, rf_max
+    # argument logic
```

Comment:
Great job on the argument logic! These are helpful error messages and I think you have covered all of the edge cases.